### PR TITLE
fix(plot): graph shifts when using the "xaxis=sim_time" option

### PR DESCRIPTION
**Pull request by tamegaictc2**
## Related links
 - ipynb
 
    [test.zip](https://github.com/tier4/CARET_analyze/files/11844334/test.zip)

## Notes for reviewers
There is no degradation using CARET_report.
CARET_report_sim_time

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Comment by tamegaictc2** [comment]
**Result of period_timeseries**

1. When sim_time=False
Screenshot from 2023-06-19 09-15-07

2. When sim_time=False, The rosbag is played with -r 0.2
Screenshot from 2023-06-19 09-15-22

3. When sim_time=False, The rosbag is played with -r 0.8
Screenshot from 2023-06-19 09-16-29

File: src/caret_analyze/plot/metrics_base.py [hunk 1]
```diff
@@ -55,16 +55,19 @@ def _convert_timeseries_records_to_sim_time(
         timeseries_records_list: list[RecordsInterface]
     ) -> list[RecordsInterface]:
         # get converter
+        records_range = Range([to.to_records() for to in self.target_objects])
```
Comment (rokamu623) [inline src/caret_analyze/plot/metrics_base.py:58]: What does the variable name `to` mean?
How about: `obj`

**Review by rokamu623 — CHANGES_REQUESTED** [review]
Reviewed by!

File: src/caret_analyze/plot/metrics_base.py (hunk 1 above)
Comment (tamegaictc2) [reply src/caret_analyze/plot/metrics_base.py:58]: I used the code from "timeseries.py" as a reference.

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Review by rokamu623 — APPROVED** [review]

**Review by atsushi421 — APPROVED** [review]
LGTM

**Review by nabetetsu — APPROVED** [review]
LGTM :+1: 
I have confirmed that the analysis results are the same when running autoware  via logging_simulator at different rates.

callback period (rate 0.2)
r02_callback_period

callback period (rate 0.8)
r08_callback_period

**Pull request merged**